### PR TITLE
Add support for Jsonnet in the Semgrep Docker image

### PR DESCRIPTION
**Pull request by aryx**
test plan:
```
$ make build-docker
$ docker run --rm -v "${PWD}:/src" semgrep:latest semgrep --config
semgrep.jsonnet
...
Support for Jsonnet rules is experimental and currently meant for internal use only. The syntax may change or be removed at any point.
Semgrep rule registry URL is https://semgrep.dev/registry.

Scanning across multiple languages:
     ocaml | 41 rules × 362 files
    python |  5 rules × 224 files
...
```

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -102,12 +102,16 @@ COPY cli ./
...
+#  - build-base: ??
+#  - make, g++: to compile the jsonnet C++ library which is installed
+#    by 'pip install jsonnet'
 # TODO? why the mkdir -p /tmp/.cache?
-# TODO: for jsonnet, you need for the build-deps before 'pip install jsonnet'
-#  'apk add g++ make'
 # hadolint ignore=DL3013
-RUN apk add --no-cache --virtual=.build-deps build-base &&\
+RUN apk add --no-cache --virtual=.build-deps build-base make g++ &&\
+     pip install jsonnet &&\
```
Comment (brendongo) [inline Dockerfile:116]: Should we add it as a dependency in setup.py or are you trying to keep this internal use only for testing for now?

Comment (brendongo) [reply Dockerfile:116]: https://github.com/returntocorp/semgrep/blob/develop/cli/setup.py#L108

Comment (aryx) [reply Dockerfile:116]: yep internal, next step would be setup.py

Comment (aryx) [reply Dockerfile:116]: I can add a comment about it

**Pull request merged**